cmd/backport-verifier/server.go

```diff
@@ -0,0 +1,188 @@
+	GetPullRequest(org, repo string, number int) (*github.PullRequest, error)
+
+	CreateComment(owner, repo string, number int, comment string) error
+
+	AddLabel(org, repo string, number int, label string) error
+	RemoveLabel(org, repo string, number int, label string) error
+}
+
+const (
+	validBackportsLabel   = "backports/valid-commits"
+	invalidBackportsLabel = "backports/invalid-commits"
+)
+
+var (
+	commandRe      = regexp.MustCompile(`(?mi)^/validate-backports\s*$`)
+	upstreamPullRe = regexp.MustCompile(`^UPSTREAM: ([0-9]+): `)
+)
+
+func helpProvider(_ []prowconfig.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: `The backport validation plugin is used to validate that backports come from merged PRs in a configured upstream repository.`,
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/validate-backports",
+		Description: "Validate that backports come from merged PRs in the upstream repository",
+		WhoCanUse:   "Anyone",
+		Examples:    []string{"/validate-backports"},
+	})
+	return pluginHelp, nil
+}
+
+type server struct {
+	config func() *Config
+
+	ghc githubClient
+}
+
+func (s *server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent) {
+	if !commandRe.MatchString(ic.Comment.Body) {
+		return
+	}
+	l.Info("Backport validation of PR has been requested.")
+	s.handle(l, ic.Repo.Owner.Login, ic.Repo.Name, ic.Comment.User.Login, ic.Issue.Number)
+}
+
+func (s *server) handlePullRequestEvent(l *logrus.Entry, event github.PullRequestEvent) {
+	if event.Action != github.PullRequestActionOpened && event.Action != github.PullRequestActionSynchronize {
+		return
+	}
+	l.Info("Changes to pull request require backport validation")
+	s.handle(l, event.Repo.Owner.Login, event.Repo.Name, event.PullRequest.User.Login, event.PullRequest.Number)
+}
+
+func (s *server) handle(l *logrus.Entry, org, repo, user string, num int) {
+	logger := l.WithFields(logrus.Fields{
+		github.OrgLogField:  org,
+		github.RepoLogField: repo,
+		github.PrLogField:   num,
+	})
+
+	upstream, configured := s.config().Repositories[fmt.Sprintf("%s/%s", org, repo)]
+	if !configured {
+		if err := s.ghc.CreateComment(org, repo, num, fmt.Sprintf("@%s: no upstream repository is configured for validating backports for this repository.", user)); err != nil {
+			logger.WithError(err).Warn("couldn't create comment")
+		}
+		ensureLabels(s.ghc, l, invalidBackportsLabel, org, repo, num)
+		return
+	}
+
+	parts := strings.Split(upstream, "/")
+	upstreamOrg, upstreamRepo := parts[0], parts[1]
+
+	commits, err := s.ghc.ListPRCommits(org, repo, num)
+	if err != nil {
+		if commentErr := s.ghc.CreateComment(org, repo, num, fmt.Sprintf(`@%s: could not list commits in this pull request. Please try again with /validate-backports.
+
+<details>
+
+%s
+
+</details>`, user, err)); commentErr != nil {
+			logger.WithError(commentErr).Warn("couldn't list commits")
```

Comment:
The point I made below and which you resolved so you don't see my response anymore still applies: We need to return if `err != nil`, not only when `commentErr != nil`, your return is a branch too high